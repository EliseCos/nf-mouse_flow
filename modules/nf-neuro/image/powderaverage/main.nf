process IMAGE_POWDERAVERAGE {
    tag "$meta.id"
    label 'process_single'

    container "scilus/scilpy:2.2.0_cpu"

    input:
        tuple val(meta), path(dwi), path(bval), path(mask)

    output:
        tuple val(meta), path("*pwd_avg.nii.gz")    , emit: pwd_avg
        path "versions.yml"                         , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"

    def b0_thr = task.ext.b0_thr ? "--b0_thr ${task.ext.b0_thr}" : ''
    def shells = task.ext.shells ? "--shells ${task.ext.shells}" : ''
    def shell_thr = task.ext.shell_thr ? "--shell_thr ${task.ext.shell_thr}" : ''

    if ( mask ) args += "--mask ${mask}"

    """
    scil_dwi_powder_average $dwi $bval ${prefix}_pwd_avg.nii.gz \
        $b0_thr $shells $shell_thr $args

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        scilpy: \$(uv pip -q -n list | grep scilpy | tr -s ' ' | cut -d' ' -f2)
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"

    """
    touch ${prefix}_pwd_avg.nii.gz

    scil_dwi_powder_average -h

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        scilpy: \$(uv pip -q -n list | grep scilpy | tr -s ' ' | cut -d' ' -f2)
    END_VERSIONS
    """
}
