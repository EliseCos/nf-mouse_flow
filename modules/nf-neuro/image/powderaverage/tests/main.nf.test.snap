{
    "powderaverage - with mask and shells": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_pwd_avg.nii.gz:md5,b0065ccd52d4921738a6538c3cd5d86a"
                    ]
                ],
                "1": [
                    "versions.yml:md5,207b645eefea02b3e70dc49c850ae68d"
                ],
                "pwd_avg": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_pwd_avg.nii.gz:md5,b0065ccd52d4921738a6538c3cd5d86a"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,207b645eefea02b3e70dc49c850ae68d"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "25.04.7"
        },
        "timestamp": "2025-09-17T20:04:21.423959602"
    },
    "powderaverage - stub-run": {
        "content": [
            [
                "versions.yml:md5,207b645eefea02b3e70dc49c850ae68d"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "25.04.7"
        },
        "timestamp": "2025-09-17T20:04:34.511995041"
    },
    "powderaverage - no mask and no shell": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_pwd_avg.nii.gz:md5,168d711e7b84fad327966ca3d49ede0f"
                    ]
                ],
                "1": [
                    "versions.yml:md5,207b645eefea02b3e70dc49c850ae68d"
                ],
                "pwd_avg": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_pwd_avg.nii.gz:md5,168d711e7b84fad327966ca3d49ede0f"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,207b645eefea02b3e70dc49c850ae68d"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "25.04.7"
        },
        "timestamp": "2025-09-17T20:04:07.648628207"
    }
}